lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -62,82 +62,50 @@ def initialize(options={})
       end
 
       def purchase(amount, payment, options = {})
-        commit("AUTH_CAPTURE") do |xml|
-          add_order_id(xml, options)
-          xml.transactionRequest do
-            xml.transactionType('authCaptureTransaction')
-            xml.amount(amount(amount))
-
-            add_payment_source(xml, payment)
-            add_invoice(xml, options)
-            add_customer_data(xml, payment, options)
-            add_market_type(xml, payment)
-            add_settings(xml, payment, options)
-            add_user_fields(xml, amount, options)
+        if payment.is_a?(String)
```

Comment:
Looked at this more and I'm thinking the existing may be simpler since in this gateway's case, there's one path that cares if it's CIM, and another path that cares about the others which aren't just cards (there's eChecks, credit cards, apple pay stuff).